netio.c

```diff
@@ -400,11 +438,27 @@ int dropbear_listen(const char* address, const char* port,
 		return -1;
 	}
 
+	/*
+	 * when listening on server-assigned-port 0
+	 * the assigned ports may differ for address families (v4/v6)
+	 * causing problems for tcpip-forward
+	 * caller can do a get_socket_address to discover assigned-port
+	 * hence, use same port for all address families
+	 */
+	u_int16_t *allocated_lport_p = 0;
+	int allocated_lport = 0;
 
 	nsock = 0;
 	for (res = res0; res != NULL && nsock < sockcount;
 			res = res->ai_next) {
 
+		if (AF_INET == res->ai_family) {
+			allocated_lport_p = &((struct sockaddr_in *)res->ai_addr)->sin_port;
```

Comment:
allocated_lport_p seems too tricky. Just go like this?
```
if allocated_port 
    if AF_INET 
        .....->sin_port = htons(...)
    else if AF_INET6 {
      .... -> sin6_port = htons(...)

```
?

Also note typo AF_INET not AF_INET6